specification/securityinsights/resource-manager/Microsoft.SecurityInsights/stable/2020-01-01/SecurityInsights.json

```diff
@@ -0,0 +1,2489 @@
+          "type": "string"
+        },
+        "properties": {
+          "$ref": "#/definitions/ActionResponseProperties",
+          "description": "Action properties for get request",
+          "x-ms-client-flatten": true
+        }
+      },
+      "type": "object"
+    },
+    "ActionResponseProperties": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/ActionPropertiesBase"
+        }
+      ],
+      "description": "Action property bag.",
+      "properties": {
+        "workflowId": {
+          "description": "The name of the logic app's workflow.",
+          "type": "string"
+        }
+      },
+      "type": "object"
+    },
+    "ActionsList": {
+      "description": "List all the actions.",
+      "properties": {
+        "nextLink": {
+          "description": "URL to fetch the next set of actions.",
+          "readOnly": true,
+          "type": "string"
+        },
+        "value": {
+          "description": "Array of actions.",
+          "items": {
+            "$ref": "#/definitions/ActionResponse"
+          },
+          "type": "array"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "AlertRule": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/ResourceWithEtag"
+        },
+        {
+          "$ref": "#/definitions/AlertRuleKind"
+        }
+      ],
+      "description": "Alert rule.",
+      "discriminator": "kind",
+      "required": [
+        "kind"
+      ],
+      "type": "object"
+    },
+    "AlertRuleKind": {
+      "description": "Describes an Azure resource with kind.",
+      "properties": {
+        "kind": {
+          "description": "The kind of the alert rule",
+          "enum": [
+            "Scheduled",
+            "MicrosoftSecurityIncidentCreation",
+            "Fusion"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "modelAsString": true,
+            "name": "AlertRuleKind",
+            "values": [
+              {
+                "value": "Scheduled"
+              },
+              {
+                "value": "MicrosoftSecurityIncidentCreation"
+              },
+              {
+                "value": "Fusion"
+              }
+            ]
+          }
+        }
+      },
+      "required": [
+        "kind"
+      ],
+      "type": "object"
+    },
+    "AlertRuleTemplate": {
+      "allOf": [
+        {
+          "$ref": "#/definitions/Resource"
+        },
+        {
+          "$ref": "#/definitions/AlertRuleKind"
+        }
+      ],
+      "description": "Alert rule template.",
+      "discriminator": "kind",
+      "required": [
+        "kind"
+      ],
+      "type": "object"
+    },
+    "AlertRuleTemplateDataSource": {
+      "description": "alert rule template data sources",
+      "properties": {
+        "connectorId": {
+          "description": "The connector id that provides the following data types",
+          "type": "string"
+        },
+        "dataTypes": {
```

Comment:
>"dataTypes" [](start = 8, length = 11)

can the possible dataTypes in this array be modeled as an enum so the user knows the options here?  Or is it too open ended?

Reply:
It got too many options and its too open ended